src/dso_api/dynamic_api/middleware.py

```diff
@@ -19,7 +19,16 @@ def __init__(self, get_response):
 
     def process_request(self, request):
         # This also installs the `user_scopes` for the OAS views.
-        request.user_scopes = UserScopes(request.GET, request.is_authorized_for, self.all_profiles)
+
+        # get_token_scopes should be set by authorization_django. We use it,
+        # instead of is_authorized_for, to get more control over authorization
+        # checks and to enable more precise logging.
+        if hasattr(request, "get_token_scopes"):
+            # get_token_scopes is a data attribute, not a method.
+            scopes = request.get_token_scopes
+        else:
+            scopes = set()
```

Comment:
Waarom niet verplichten dat deze er gewoon is? Het ontbreken ervan betekend toch een configuratiefout in onze app.

Reply:
Een van de tests faalde zonder deze hack.

Reply:
Done.

Reply:
Sorry, toch niet done, ik deed iets verkeerd. Het lijkt erop dat de requests die uit `api_client` komen geen `get_token_scopes` hebben.

Reply:
@vdboor Ik ben erachter wat hier gebeurt. Een OPTIONS-request heeft geen `get_token_scopes`.